src/utils/api.js

```diff
@@ -88,20 +88,28 @@ export const getPeriodTypes = () =>
 /**
  * @return {Promise} Organisation units
  */
-export const getOrganisationUnits = () =>
-    d2.models.organisationUnits
-        .list({
+export const getOrganisationUnits = async () => {
+    const orgUnitFields = [
+        'id',
+        'displayName',
+        'path',
+        'children::isNotEmpty',
+        'memberCount',
+    ].join(',')
+    const organisationUnits = await d2.currentUser.getOrganisationUnits({
+        fields: orgUnitFields,
+    })
+    if (organisationUnits.size > 0) {
+        return organisationUnits.toArray()
+    } else {
+        const orgUnitModel = await d2.models.organisationUnits.list({
             paging: false,
             level: 1,
```

Comment:
I'm not entirely sure, but I think the new approach combined with `level: 1` could be problematic, because the root node for a given user might be at a lower level (i.e. higher value for the level param). I'm fairly certain you can remove the `level` param.

To verify, you could create a user with access to the reports-app and assign the user to two lower level org-units. My expectation is that the tree will be empty in the current implementation, and the tree would look correct after removing the `level` param.

Reply:
The new approach of getting the user's assigned org units using `d2.currentUser.getOrganisationUnits` does not provide the `level` param, only the `fields` param.

While the system would ideally not return any org units for level 1 if the user does not have acces to them, the current API response does return them and this is the root of issue https://jira.dhis2.org/browse/DHIS2-11537. This is shown in the table below:

| User has assigned org units | Existing (now fallback) behaviour |
|---|---|
| ![image](https://user-images.githubusercontent.com/4295266/131670793-0baa7a28-fa9d-4a03-b6fe-39267a95e7cd.png) | ![image](https://user-images.githubusercontent.com/4295266/131670955-f9697f6e-b303-4972-ae5d-5f7bebd32f68.png) |

When omitting `level: 1` in the fallback case, the server returns every organisation unit, which results in slow loading and rendering.